extensions/http-curl/client/HTTPClient.cpp

```diff
@@ -204,7 +264,9 @@ const char *HTTPClient::getContentType() {
 }
 
 const std::vector<char> &HTTPClient::getResponseBody() {
```

Comment:
c++11+ guarantees std::string bytes are sequential. May not need a vector<char> in lieu of string.

Reply:
Not in these changes, but wanted to note...

Reply:
yes and it changes again in 17...some not for the best unfortunately.

herb sutter has some really good talks about small string optimization and how you shouldn't fight it, but I'm not too certain we can really avoid the pessimization.  what are your thoughts on the SOO causing pessimization against strings that are larger than the buffer on things like sizeof, etc? I've seen profiles where it can really cause unnecessary havoc.